src/VolumeDrawable.ts

```diff
@@ -10,14 +10,25 @@ import { Bounds, FuseChannel } from "./types";
 import { ThreeJsPanel } from "./ThreeJsPanel";
 import { Light } from "./Light";
 import Channel from "./Channel";
+import { Pane } from "tweakpane";
+
+type ColorArray = [number, number, number];
+type ColorObject = { r: number; g: number; b: number };
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type WithObjectColors<T extends Record<string, any>> = {
+  [K in keyof T]: T[K] extends ColorArray | undefined ? ColorObject : T[K];
+};
+
+export const colorArrayToObject = ([r, g, b]: ColorArray): ColorObject => ({ r, g, b });
+export const colorObjectToArray = ({ r, g, b }: ColorObject): ColorArray => [r, g, b];
 
 // A renderable multichannel volume image with 8-bits per channel intensity values.
 export default class VolumeDrawable {
   public PT: boolean;
   public volume: Volume;
   private onChannelDataReadyCallback?: () => void;
-  private translation: Vector3;
-  private rotation: Euler;
+  public translation: Vector3;
+  public rotation: Euler;
```

Comment:
do these need to be public because of `pane.addInput(this, "translation")` ?

Reply:
Right, `addInput` has a type constraint that the second argument must be a `keyof` the first, and `"translation"` doesn't appear in the interface when it's private (even within its own method!).